applications/serial_lte_modem/src/slm_at_fota.c

```diff
@@ -375,15 +435,67 @@ void slm_fota_post_process(void)
 
 void slm_finish_modem_fota(int modem_lib_init_ret)
 {
-	if (handle_nrf_modem_lib_init_ret(modem_lib_init_ret)) {
-		char buf[40];
+	char buf[40];
+#if defined(CONFIG_SLM_FULL_FOTA)
+	int err;
+
+	ARG_UNUSED(modem_lib_init_ret);
+
+	if (fota_type == DFU_TARGET_IMAGE_TYPE_FULL_MODEM) {
+		/* Full fota activation differs from delta modem fota. */
+		LOG_INF("Applying full modem firmware update from external flash\n");
+
+		err = nrf_modem_lib_shutdown();
+		if (err != 0) {
+			LOG_ERR("nrf_modem_lib_shutdown() failed: %d\n", err);
+			return;
+		}
+
+		err = nrf_modem_lib_bootloader_init();
+		if (err != 0) {
+			LOG_ERR("nrf_modem_lib_bootloader_init() failed: %d\n", err);
```

Comment:
If we make reset here, it has to delayed one because we have to return error event. Sounds quite complicated

Reply:
After FOTA and reset/modemreset, slm_fota_post_process() function is called. If the FOTA activation fails, slm_fota_post_process function sends an error AT event.

Reply:
Yes, this might work. I have to simulate a finalization error and test. There are actually two different use cases to test, modem reset and device reset.

Reply:
Error handling of the full FOTA activation has been changed.